plugins/Marketplace/vue/src/Marketplace/Marketplace.vue

```diff
@@ -23,251 +23,243 @@
-          // and via testing in browser
-          //
-          // a) visible downloads count
-          //    -> clamp to 2 lines if title is 2 lines or more or alert is 2 lines or more
-          //       or together are more than 3 lines
-          //    -> clamp to 1 line if title is over 2 lines and alert is over 2 lines simultaneously
-          // b) no downloads count (i.e. a premium plugin)
-          //    -> clamp to 2 lines if sum of lines for title and notification is over 4
-          if (hasDownloads) {
-            if ((titleLines >= 2 || alertLines > 2) || (titleLines + alertLines >= 4)) {
-              clampedLines = 2;
-            }
-            if (titleLines + alertLines >= 5) {
-              clampedLines = 1;
-            }
-          } else if (titleLines + alertLines >= 5) {
-            clampedLines = 2;
-          }
+      if (!doFetch) {
+        return;
+      }
 
-          if (clampedLines) {
-            cardDescription.setAttribute('data-clamp', `${clampedLines}`);
-          } else {
-            cardDescription.removeAttribute('data-clamp');
-          }
-        }
+      this.fetchPlugins();
+    },
+    updateQuery(event: Event) {
+      MatomoUrl.updateHash({
+        ...MatomoUrl.hashParsed.value,
+        query: event,
       });
-    }, 100);
-
-    nextTick(() => {
-      const cardSelector = '.marketplace .card-holder';
+    },
+    updateType(event: Event) {
+      MatomoUrl.updateHash({
+        ...MatomoUrl.hashParsed.value,
+        pluginType: event,
+      });
+    },
+    updateSort(event: Event) {
+      MatomoUrl.updateHash({
+        ...MatomoUrl.hashParsed.value,
+        sort: event,
+      });
+    },
+    fetchPlugins() {
+      this.loading = true;
+      this.pluginsToShow = [];
 
-      addCardClickHandler(cardSelector);
-      shrinkDescriptionIfMultilineTitle(cardSelector);
+      if (this.fetchRequestAbortController) {
+        this.fetchRequestAbortController.abort();
+        this.fetchRequestAbortController = null;
+      }
 
-      $(window).resize(() => {
-        shrinkDescriptionIfMultilineTitle(cardSelector);
-      });
-    });
-  },
-  methods: {
-    changePluginSort() {
-      MatomoUrl.updateUrl(
+      this.fetchRequestAbortController = new AbortController();
+      this.fetchRequest = AjaxHelper.post(
         {
-          ...MatomoUrl.urlParsed.value,
-          query: '',
-          sort: this.pluginSort,
+          module: 'Marketplace',
+          action: 'searchPlugins',
+          format: 'JSON',
         },
         {
-          ...MatomoUrl.hashParsed.value,
-          query: '',
+          query: this.searchQuery,
           sort: this.pluginSort,
-        },
-      );
-    },
-    changePluginType() {
-      MatomoUrl.updateUrl(
-        {
-          ...MatomoUrl.urlParsed.value,
-          query: '',
-          show: this.pluginTypeFilter,
+          themesOnly: this.pluginTypeFilter === 'themes',
```

Comment:
would this work here?
```suggestion
          themesOnly: this.showThemes,
```